tests/Integration/classes/module/ModuleBlockReassuranceTest.php

```diff
@@ -0,0 +1,194 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Integration;
+
+use PrestaShop\PrestaShop\Core\Addon\Module\ModuleManagerBuilder;
+use Tests\TestCase\IntegrationTestCase;
+use Module;
+
+/**
+ * Test that module blockreassurance behavior is correct
+ *
+ * Is tested:
+ * - install process
+ * - uninstall process
+ * Tests to add:
+ * - rendering of the widget (cannot be tested now, it requires proper Smarty initialization)
+ * - rendering of thr admin form (cannot be tested now, it requires proper Smarty initialization)
+ */
+class ModuleBlockReassuranceTest extends IntegrationTestCase
+{
+    /**
+     * Flag used to control whether the module should be uninstalled at tear down after a test
+     *
+     * @var bool
+     */
+    private static $shouldUninstallAtTearDown = true;
+
+    protected function setUp()
+    {
+        $moduleLocation = _PS_MODULE_DIR_ . '/blockreassurance';
+
+        $doesModuleExistIntoTestModuleDirectory = ((file_exists($moduleLocation)) && (is_dir($moduleLocation)));
+
+        // copy module from project into tests
+        if (false === $doesModuleExistIntoTestModuleDirectory) {
+            $expectedModuleLocation = _PS_ROOT_DIR_ . '/modules/blockreassurance';
+            $doesModuleExistIntoPSModuleDirectory = ((file_exists($expectedModuleLocation)) && (is_dir($expectedModuleLocation)));
+
+            if (false === $doesModuleExistIntoPSModuleDirectory) {
+                self::$shouldUninstallAtTearDown = false;
+                throw new \Exception('Looks like module blockreassurance is not installed. It must be for the test to run. Run $ composer install.');
+            }
+
+            $copyResult = self::copyr($expectedModuleLocation, $moduleLocation);
+
+            if (false === $copyResult) {
+                throw new \Exception('Failed to copy blockreassurance module for tests.');
+            }
+        }
+
+        parent::setUp();
+    }
+
+    /**
+     * Test if blockreassurance module install is successfull
+     */
+    public function testInstallIsSuccessfull()
+    {
+        $moduleManager = ModuleManagerBuilder::getInstance()->build();
+        $moduleManager->install('blockreassurance');
+        $module = Module::getInstanceByName('blockreassurance');
+        \Cache::clean('hook_alias');
+
+        // assert install is successfull
+
+        self::assertTrue($moduleManager->isInstalled('blockreassurance'));
```

Comment:
use `$this` instead of `self::`  to do the same thing everywhere, even if it's declared static :/ Otherwise we need to plan to change it everywhere

Reply:
this is already what we are doing :+1:, so I'm in favor of self